OSC rdma warnings

In master using gcc-7.3:
```shell
In file included from osc_rdma_comm.h:20:0,
                 from osc_rdma_accumulate.c:17:
osc_rdma_accumulate.c: In function 'ompi_osc_rdma_acc_single_atomic':
osc_rdma_lock.h:85:12: warning: 'ret' may be used uninitialized in this function [-Wmaybe-uninitialized]
         if (OPAL_LIKELY(1 == ret)) {
            ^
osc_rdma_lock.h:43:9: note: 'ret' was declared here
     int ret;
```

--- a/ompi/mca/osc/rdma/osc_rdma_lock.h
+++ b/ompi/mca/osc/rdma/osc_rdma_lock.h
@@ -40,7 +40,7 @@ static inline int ompi_osc_rdma_btl_fop (ompi_osc_rdma_module_t *module, struct 
                                          ompi_osc_rdma_pending_op_cb_fn_t cbfunc, void *cbdata, void *cbcontext)
 {
     ompi_osc_rdma_pending_op_t *pending_op;
-    int ret;
+    int ret = OPAL_ERROR;
 
     pending_op = OBJ_NEW(ompi_osc_rdma_pending_op_t);
     assert (NULL != pending_op);

